src/NzbDrone.Core/MediaFiles/MediaFileDeletionService.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using NLog;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Exceptions;
+using NzbDrone.Core.Messaging.Events;
+using NzbDrone.Core.Tv;
+using NzbDrone.Core.Tv.Events;
+
+namespace NzbDrone.Core.MediaFiles
+{
+    public interface IDeleteMediaFiles
+    {
+        void DeleteEpisodeFile(Series series, EpisodeFile episodeFile);
+    }
+
+    public class MediaFileDeletionService : IDeleteMediaFiles, IHandleAsync<SeriesDeletedEvent>
+    {
+        private readonly IDiskProvider _diskProvider;
+        private readonly IRecycleBinProvider _recycleBinProvider;
+        private readonly IMediaFileService _mediaFileService;
+        private readonly Logger _logger;
+
+        public MediaFileDeletionService(IDiskProvider diskProvider,
+                                        IRecycleBinProvider recycleBinProvider,
+                                        IMediaFileService mediaFileService,
+                                        Logger logger)
+        {
+            _diskProvider = diskProvider;
+            _recycleBinProvider = recycleBinProvider;
+            _mediaFileService = mediaFileService;
+            _logger = logger;
+        }
+
+        public void DeleteEpisodeFile(Series series, EpisodeFile episodeFile)
+        {
+            var fullPath = Path.Combine(series.Path, episodeFile.RelativePath);
+            var rootFolder = _diskProvider.GetParentFolder(series.Path);
+
+            if (!_diskProvider.FolderExists(rootFolder))
+            {
+                throw new NzbDroneClientException(HttpStatusCode.NotFound, "Series' root folder ({0}) doesn't exist.", rootFolder);
```

Comment:
If the episodefile doesn't exist in the db at all then 404. Otherwise I'm leaning to Conflict, we already use it in the ReleaseModule if adding the release failed. But we need something that shows up on UI as a proper error notification.